e2etests/node_scan_rhcos_test.go

```diff
@@ -0,0 +1,186 @@
+//go:build e2e
+// +build e2e
+
+package e2etests
+
+import (
+	"context"
+	"testing"
+
+	v1 "github.com/stackrox/scanner/generated/scanner/api/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// vulnLibksba is an example of a fixable vulnerability
+var vulnLibksba = &v1.Vulnerability{
+	Name:        "RHSA-2022:7089",
+	Description: "KSBA (pronounced Kasbah) is a library to make X.509 certificates as well as the CMS easily accessible by other applications.  Both specifications are building blocks of S/MIME and TLS.\n\nSecurity Fix(es):\n\n* libksba: integer overflow may lead to remote code execution (CVE-2022-3515)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.",
+	Link:        "https://access.redhat.com/errata/RHSA-2022:7089",
+	MetadataV2: &v1.Metadata{
+		PublishedDateTime:    "2022-10-24T00:00Z",
+		LastModifiedDateTime: "",
+		CvssV2:               nil,
+		CvssV3: &v1.CVSSMetadata{
+			Score:               8.1,
+			Vector:              "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
+			ExploitabilityScore: 2.2,
+			ImpactScore:         5.9,
+		},
+	},
+	FixedBy:  "0:1.3.5-8.el8_6",
+	Severity: "Important",
+}
+
+// vulnTar is an example of a non-fixable vulnerability
+var vulnTar = &v1.Vulnerability{
+	Name:        "CVE-2005-2541",
+	Description: "DOCUMENTATION: The MITRE CVE dictionary describes this issue as: Tar 1.15.1 does not properly warn the user when extracting setuid or setgid files, which may allow local users or remote attackers to gain privileges. \n            STATEMENT: This CVE was assigned to what is the documented and expected behaviour of tar.  There are currently no plans to change tar behaviour to strip setuid and setgid bits when extracting archives.",
+	Link:        "https://access.redhat.com/security/cve/CVE-2005-2541",
+	MetadataV2: &v1.Metadata{
```

Comment:
Is it ok that `PublishedDateTime` is absent here or would it only be present for a fixable vuln?

Reply:
Need to check that...

Reply:
In this case, scanner is missing the `PublishedDateTime`. The vulnerability metadata looks like this:
`metadata_v2:<cvss_v3:<vector:"CVSS:3.1/AV:L/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H" score:7 exploitability_score:1 impact_score:5.9>>`.
I am assuming that we cannot rely on this field being set all the time.

Reply:
I'm fine either way.